rct229/rules/section5/section5rule49.py

```diff
@@ -0,0 +1,132 @@
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.std_comparisons import std_equal
+from rct229.schema.config import ureg
+
+TARGET_AIR_LEAKAGE_COEFF = 0.6 * ureg("cfm / foot**2")
+
+
+class Section5Rule49(RuleDefinitionListIndexedBase):
+    """Rule 49 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule49, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule49.BuildingRule(),
+            index_rmr="proposed",
+            id="5-49",
+            description="The proposed air leakage rate of the building envelope (I75Pa) at a fixed building pressure differential of 0.3 in. of water shall be 0.6 cfm/ft2 for buildings providing verification in accordance with Section 5.9.1.2. The air leakage rate of the building envelope shall be converted to appropriate units for the simulation program using one of the methods in Section G3.1.1.4. Exceptions: When whole-building air leakage testing, in accordance with Section 5.4.3.1.1, is specified during design and completed after construction, the proposed design air leakage rate of the building envelope shall be as measured..",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_proposed = context.proposed
+        return {"climate_zone": rmr_proposed["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule49.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                required_fields={
+                    "$..zones[*]": ["infiltration", "surfaces"]
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            building_p = context.proposed
+
+            scc_dict_p = get_surface_conditioning_category_dict(
+                data["climate_zone"], building_p
+            )
+            zcc_dict_p = get_zone_conditioning_category_dict(
+                data["climate_zone"], building_p
+            )
+
+            building_total_envelope_area = 0.0
+            building_total_air_leakage_rate = 0.0
+            building_total_measured_air_leakage_rate = 0.0
+            empty_measured_air_leakage_rate_flow_flag = False
+
+            building_total_envelope_area = sum(
+                [
+                    surface.get("area")
+                    for surface in find_all("$..surfaces[*]", building_p)
+                    if scc_dict_p[surface["id"]] != SCC.UNREGULATED
+                ],
+                ZERO.AREA,
+            )
+
+            for zones in find_all("$..zones[*]", building_p):
+                if zones["infiltration"].get("measured_air_leakage_rate"):
```

Comment:
Instead, I used ```measured_air_leakage_rate = zone["infiltration"].get(
                    "measured_air_leakage_rate"
                )```
because there could be a case when ```measured_air_leakage_rate``` doesn't exist and this case caused an error.  I'd appreciate it if you would review this when I re-request this PR review.